### PR TITLE
Commit within

### PR DESCRIPTION
**Pull request by atz**
Want to get this merged before I get too far afield.  Note the massive (and unbreaking!) simplification in routes.

### CODE REVIEW

File: spec/helpers/registration_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'spec_helper'
+
+describe RegistrationHelper do
+  it 'does something' do
```
Comment (mejackreed) [inline spec/helpers/registration_helper_spec.rb]: Is this file a placeholder?

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -96,17 +94,12 @@ def remove_collection
   end
 
   def register
-    @perm_keys = ["sunetid:#{current_user.login}"]
-    if webauth && webauth.privgroup.present?
-      @perm_keys += webauth.privgroup.split(/\|/).collect { |g| "workgroup:#{g}" }
-    end
-    render :register, :layout => 'application'
+    @perm_keys = current_user.groups
```
Comment (drh-stanford) [inline app/controllers/items_controller.rb]: is there a test for this logic change?

**Comment by drh-stanford** [comment]
i'm :+1: but the multiple commits could be squashed into a cleanup commit

File: spec/helpers/registration_helper_spec.rb (hunk 1 above)
Comment (atz) [inline spec/helpers/registration_helper_spec.rb]: Yup.  I complained about there not being even a placeholder for the stuff being untested.  So I added that.

File: app/controllers/items_controller.rb (hunk 2 above)
Comment (atz) [inline app/controllers/items_controller.rb]: Covered by existed tests. The old code is logically identical (copy/pasted) from the code under `groups`.

**Comment by atz** [comment]
I squashed several commits and also added a few more that are distinct (related to GMT requirement of HTTP spec).

File: app/views/items/file.html.erb [hunk 3]
```diff
@@ -1,24 +1,22 @@
...
-
-  if workspace_files.include?(params[:file])
-    link='/items/'+params[:id]+'/file/?file='+params[:file]
+  # @NOTE: list_files performs remote SSH connection, may fail!
+  # @TODO: do list_files inside app and pass files to template for render, or ideally make it async
+  if @object.list_files.include?(params[:file])
+    link = "/items/#{params[:id]}/file/?file=#{params[:file]}"
     %>
-    Workspace: <a href=<%= link%>><%=params[:file]%><a/>
+    Workspace: <a href="<%= link %>"><%= params[:file] %><a/>
```
Comment (jmartin-sul) [inline app/views/items/file.html.erb]: i think maybe the end should be `</a>`?

Comment (cbeer) [inline app/views/items/file.html.erb]: Or, while you're at it, `#link_to`?

**Pull request merged**